frontend/src/modules/displays/MultiStepWizard.tsx

```diff
@@ -363,6 +366,15 @@ const MultiStepWizard = (): JSX.Element => {
       const param = multiParams.find((param: Data) => param.stateKey === parameter.stateKey);
       if (param) param.alarmValues = [min, max];
       parameter.alarmValues = [min, max];
+      if (location.pathname === DASHBOARD_ROUTE.path) {
```

Comment:
What's going on here, and what is it for?

Reply:
so the bug that dispatch caused is:
- go to set alarms page, change any one of the alarm limits to anything, basically unsaved changes
- open alarms modal, (eg: HR too high) and change alarm limits for HR and press `cancel`
- pressing cancel will discard the changes, which would mean that the unsaved changes on the alarms page also gets discarded

Reply:
basically, we were dispatching only for HFNC control to show the unsaved alarm limit changes so maybe it's safe to do this check?

Reply:
I see, it'd be great if you could add a code comment summarizing this. What happens if we open the breadcrumbs modal from the event log while we're on the settings page, and try to do the same test?

Reply:
Sure will do that!

Reply:
> I see, it'd be great if you could add a code comment summarizing this. What happens if we open the breadcrumbs modal from the event log while we're on the settings page, and try to do the same test?

Seems to be fine on light testing.

Reply:
Ok. The specific outcome I'd look for in that test case is that you press cancel to discard the changes from the breadcrumbs modal while on the Settings screen; and then if you go to the Alarm Limits screen or open an Alarm Limits modal, does it load with the unsaved changes (which would be undesirable)? That's what I'd expect to happen based on my interpretation of your explanation for this code.

Reply:
I think the primary issue here is unsaved changes in the alarms page and unsaved changes in breadcrumbs modal are different, or rather unsaved changes in alarms page are not propagated to breadcrumbs modal ( they probably look at current/set alarm limits), so when I say I'm discarding unsaved changes in breadcrumbs modal it means the temporary changes made there, which are currently different from the changes on alarms page, and discarding temporary changes in breadcrumbs should not discard the `different` changes on alarms page from my understanding. I'll merge for now and keep this unresolved.

Reply:
Oh ok, so the breadcrumbs modal isn't loading/saving alarm limits changed from/to `store.controller.alarmLimits.draft`? That would explain why it behaves the way it does, then. But then how is the MultiStepWizard's top-level view (which displays all alarm limits and settings, including unsaved changes to alarm limits) showing the unsaved changes?

Reply:
> Oh ok, so the breadcrumbs modal isn't loading/saving alarm limits changed from/to `store.controller.alarmLimits.draft`?

I think so.

> But then how is the MultiStepWizard's top-level view (which displays all alarm limits and settings, including unsaved changes to alarm limits) showing the unsaved changes?

Do you mean the HFNC control? if so then it uses a selector to show changes.

Reply:
Yes, HFNC control. So here's my interpretation of what you're saying:

1. The breadcrumbs modal stores its unsaved changes into local state, not into store.controller.alarmLimits.draft (looking with a fresh mind at the code in MultiStepWizard.tsx, I see where this happens, and now I see where it gets committed to store.controller.alarmLimits when a submit is confirmed, and where it gets reset when a discard is confirmed)
2. The "HFNC Control" screen in the breadcrumbs modal shows unsaved changes for alarm limits by getting them from the selector for store.controller.alarmLimits.draft, not by checking the local state (looking at the code for MultiStepWizard.tsx, it looks like this is done from the HFNCControls component's ValueInfo components, which don't have access to the local state of the breadcrumbs modal)

It looks like the dispatch you are making here is so that the local changes get copied into store.controller.alarmLimits.draft (but only when we're on the dashboard) so that unsaved changes become visible via the selector used by ValueInfo. Now I realize that I didn't fully understand your first two comments. So if I'm understanding everything correctly, then the HFNCControls screen will not show the unsaved changes if alarm limits from the breadcrumbs modal if the user is outside the dashboard screen - this would be consistent with the comment you recently added to #308. This seems like buggy behavior. Perhaps the more robust solution is to allow specifying the source of the alarm limits as props on ValueInfo and HFNCControls, so that MultiStepWizard can pass its local state into them. This should completely remove the need for dispatching unsaved changes of alarm limits into store.controller.alarmLimits.draft. We can test whether this buggy behavior exists as I'm predicting, and if so how to fix it, as part of #338.

Reply:
Yes the bug exists, i was more concerned about not discarding unsaved changes in alarms page, which was the wrong approach.

> Perhaps the more robust solution is to allow specifying the source of the alarm limits as props on ValueInfo and HFNCControls, so that MultiStepWizard can pass its local state into them. This should completely remove the need for dispatching unsaved changes of alarm limits into store.controller.alarmLimits.draft. We can test whether this buggy behavior exists as I'm predicting, and if so how to fix it, as part of #338.

Yes I agree that it's a robust approach, but I feel since we are already using selectors in ValueInfo it's still a good approach to try and fix dispatching with conditions, if we pass the local state then we might need to refactor ValueInfo to not use selectors in that case? I have opened a PR to fix the bug introduced in #338 and also the same for parameters using a dispatch and selector for ParametersRequestDraft, I have only lightly tested it so far, if it still causes unwanted regressions i will work on the approach you suggested.